core/src/main/scala/kafka/common/InterBrokerSendThread.scala

```diff
@@ -78,6 +77,9 @@ abstract class InterBrokerSendThread(
       failExpiredRequests(now)
       unsentRequests.clean()
     } catch {
+      case _: DisconnectException if isShutdownInitiated =>
+        // DisconnectException is caused by NetworkClient#ensureActive
+        // this thread is closing so this error is acceptable
```

Comment:
I am not sure about this. We would ideally drain the existing requests (up to a timeout), not accept new ones and then close the network client. This change seems like it would hide the problem, but not necessarily fix it. Let's see what @dajac and/or @hachikuji say.

Reply:
`DisconnectException` is indeed expected after `initiateClose` is called. It seems OK to handle it here but I would rather use `client.active()` instead of `isShutdownInitiated`. This is something that we should have done in https://github.com/apache/kafka/pull/9999.

@ijuma I think that your point is legitimate. It seems that we brutally stop the send thread without really draining the inflight requests. This maybe OK as we stop this thread only when the broker is shutdown anyway. In order to restore the CI, I suggest to proceed with the minor fix and to review/address this point in a followup PR if necessary.

Reply:
@dajac Thanks for your response.

>  I would rather use client.active() instead of isShutdownInitiated. This is something that we should have done in #9999.

It seems to me the exception is available only if the thread is going to shutdown. If network client get closed and it is not caused by calling Thread#shutdown, we still throw fatal exception as it is totally unexpected behavior.

Reply:
@dajac Could you take a look at above comment? thanks.

Reply:
@chia7712 I understand your point. I was thinking that the exception is actually expected when `networkClient.initiateClose()` is called so it makes sense to rely on the `networkClient` to decide if it is expected or not. Today, this is tight to the thread being shutdown as you pointed out. If we would change this in the future, it will be really easy to mis changing this condition and this could led to a subtile bug like this one. I don't feel strong for either approaches so I leave it up to you.

Reply:
>  If we would change this in the future, it will be really easy to mis changing this condition and this could led to a subtile bug like this one

make sense :)

Reply:
@chia7712 Could we file a Jira to follow up on @ijuma's point? I think that we should review this but it is fine to do this separately from this one.

Reply:
sure. see https://issues.apache.org/jira/browse/KAFKA-12356